src/configparser/mod.rs

```diff
@@ -0,0 +1,58 @@
+pub mod challenge;
+pub mod config;
+
+use anyhow::{anyhow, Error, Result};
+use itertools::Itertools;
+use simplelog::*;
+use std::sync::OnceLock;
+
+pub static CONFIG: OnceLock<config::RcdsConfig> = OnceLock::new();
+pub static CHALLENGES: OnceLock<Vec<challenge::ChallengeConfig>> = OnceLock::new();
+// type aliases for above's lifetimes
+pub type RcdsConfig = &'static config::RcdsConfig;
+pub type ChallengeConfigs = &'static Vec<challenge::ChallengeConfig>;
+
+/// get config from global, or load from file if not parsed yet
+pub fn get_config() -> Result<RcdsConfig> {
+    // return already parsed value
+    if let Some(existing) = CONFIG.get() {
+        return Ok(existing);
+    }
+
+    let config = config::parse();
+
+    // if config parsed OK, set global and return that
+    // otherwise pass through the errors from parsing
+    config.map(|c| CONFIG.get_or_init(|| c))
+}
```

Comment:
> Also refactors how config parsing works to only parse once, and reuse that on other calls.

Currently, this is with an getter function that updates a static variable. This is redundant to have both exposed, and I'd prefer to only have the static variable lazily init'd, but `std::Lazy` isnt standardized yet and would need another crate for that (`std::OnceCell` is). 

Thoughts on how this should be exposed? What's the idiomatic way to do this?